core/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -19,12 +19,198 @@
 
 package org.elasticsearch.monitor.os;
 
-/**
- *
- */
-public interface OsProbe {
+import org.apache.lucene.util.Constants;
+import org.elasticsearch.common.component.AbstractComponent;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.lang.reflect.Method;
+
+public class OsProbe extends AbstractComponent {
+
+    private static final OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
+
+    private static final Method getFreePhysicalMemorySize;
+    private static final Method getTotalPhysicalMemorySize;
+    private static final Method getFreeSwapSpaceSize;
+    private static final Method getTotalSwapSpaceSize;
+    private static final Method getSystemLoadAverage;
+
+    static {
+        Method method = null;
+        try {
+            method = osMxBean.getClass().getDeclaredMethod("getFreePhysicalMemorySize");
+            method.setAccessible(true);
+
+        } catch (Exception e) {
+            // not available
+
+        }
+        getFreePhysicalMemorySize = method;
```

Comment:
can we have a helper method that gets the method and makes it accessible? and then have `getFreePhysicalMemorySize = getMethod("getFreePhysicalMemorySize ")`. This will simplify the code, and also fix the bug here where if we fail one of the following code fails, we will assign the same method to a different method handler